b+tree.ts

```diff
@@ -488,6 +488,213 @@ export default class BTree<K=any, V=any> implements ISortedMapF<K,V>, ISortedMap
     return {nodequeue, nodeindex, leaf:nextnode};
   }
 
+  /**
+   * Computes the delta operations (adds, deletes, and modifications) required to go from this tree to the supplied destination.
+   * For efficiency, the delta is returned via invocations of supplied handlers.
+   * The computation is optimized for the case in which the two trees have large amounts of shared data (obtained by calling the
+   * `clone` or `with` APIs) and will avoid any iteration of shared state.
+   * @param destination The tree to compute a delta to.
+   * @param processAdd Callback invoked for all adds in the delta.
```

Comment:
We don't care for our usecase, but in general its possible to optimize this algorithm even more if we know which of the 3 kinds of edits the the caller wants. Specially if a caller only cared about modify, we could delta a tiny tree to a big tree in O(1) time instead of O(N*log(N)) if we knew we didn't need to call processAdd O(N) times.

Thus I think these handlers should be optional: even if we don't implement the optimization, someone might want to layer, and it will be a nicer API anyway. Maybe use an object/keyword arguments?

Another approach for this optimization could be done by passing an iterator into processAdd and processDelete. I think thats more complex though